### PR TITLE
Json save file

### LINKED ISSUE #99: Improve Json Storage
implement Json save files to differentiate teachers and students clearly

### PR DESCRIPTION
**Pull request by amoschee**
Fixes #99

### CODE REVIEW

**Comment by amoschee** [comment]
Future refactoring to consider:
make JsonAdaptedPerson abstract and have separate JsonAdapted for Student and Teacher

**Comment by amoschee** [comment]
NOTE TO OTHERS:
merge other pending PRs first before this one to minimise merge conflicts

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -106,12 +107,14 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
...
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
+        DaysAttended updatedDaysAttended = editPersonDescriptor.getDaysAttended().orElse(personToEdit
+            .getDaysAttended());
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
         Set<Subject> updatedSubjects = editPersonDescriptor.getSubjects().orElse(personToEdit.getSubjects());
         Set<String> updatedClasses = editPersonDescriptor.getClasses().orElse(personToEdit.getClasses());
 
-        return new Person(updatedName, updatedGender, updatedPhone, updatedEmail, updatedAddress, updatedTags,
-            updatedSubjects, updatedClasses);
+        return Person.createPerson(personToEdit.getType(), updatedName, updatedGender, updatedPhone, updatedEmail,
```
Comment (AgentHagu) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:116]: Nice use of a factory method to generalize creation of either Student or Teacher

**Comment by amoschee** [comment]
> Future refactoring to consider: make JsonAdaptedPerson abstract and have separate JsonAdapted for Student and Teacher

This has been implemented

File: src/main/java/seedu/address/model/person/Teacher.java [hunk 2]
```diff
@@ -97,4 +93,23 @@ public int hashCode() {
...
+     * This defines a weaker notion of equality between two persons.
+     */
+    @Override
+    public boolean isSamePerson(Person otherPerson) {
+        if (otherPerson == this) {
+            return true;
+        }
+
+        return otherPerson instanceof Teacher
+            && otherPerson.getName().equals(getName()) && otherPerson.getEmail().equals(getEmail());
```
Comment (AgentHagu) [inline src/main/java/seedu/address/model/person/Teacher.java:107]: Good and appropriate checks for identical entries

**Review by AgentHagu — APPROVED** [review]
LGTM

**Pull request merged**